qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -215,6 +209,35 @@ def _on_config_changed(self, option):
             self.maybe_hide()
         elif option == 'statusbar.padding':
             self._set_hbox_padding()
+        elif option == 'statusbar.widgets':
+            self._draw_widgets()
+
+    def _draw_widgets(self):
+        """Draw statusbar widgets."""
+        # Start with widgets hidden and show them when needed
+        for widget in [self.url, self.percentage,
+                       self.backforward, self.tabindex]:
+            self._hbox.removeWidget(widget)
+            widget.hide()
+
+        # Read the list and set widgets accordingly
+        for segment in config.val.statusbar.widgets:
+            if segment == 'url':
+                self._hbox.addWidget(self.url)
+                self.url.show()
+            elif segment == 'scroll':
+                self._hbox.addWidget(self.percentage)
+                self.percentage.show()
+            elif segment == 'scroll_raw':
+                self._hbox.addWidget(self.percentage)
+                self.percentage.raw = True
+                self.percentage.show()
+            elif segment == 'history':
+                self._hbox.addWidget(self.backforward)
+                self.backforward.show()
+            elif segment == 'tabs':
+                self._hbox.addWidget(self.tabindex)
+                self.tabindex.show()
```

Comment:
Please add an `else:` here doing something like `raise ValueError("Unknown segment {}".format(segment))` so we don't have segments which are accidentally unhandled when we add more in the future.

Reply:
I used `valid_values` in the config to avoid this, raising an exception makes qutebrowser think it crashed.